internal/providers/docker/components/container/image.go

```diff
@@ -34,7 +34,10 @@ func (c *Container) ensureImage(ctx context.Context) error {
 	c.State.Image.Command = inspection.Config.Cmd
 	c.State.Image.WorkingDir = inspection.Config.WorkingDir
 	c.State.Image.Entrypoint = inspection.Config.Entrypoint
-
+	c.State.Image.Shell = inspection.Config.Shell
+	if len(c.State.Image.Shell) == 0 {
+		c.State.Image.Shell = []string{"/bin/sh", "-c"}
```

Comment:
Looking at the code for docker engine and I see this:

```go
func defaultShellForOS(os string) []string {
	if os == "linux" {
		return []string{"/bin/sh", "-c"}
	}
	return []string{"cmd", "/S", "/C"}
}
```

I believe the OS is available in `inspection.Os`

Reply:
Made the change but I'm not a big fan of adding code I can't meaningfully test.